backend/clickhouse/sessions.go

```diff
@@ -589,3 +590,108 @@ func (client *Client) SessionsKeyValues(ctx context.Context, projectID int, keyN
 func (client *Client) GetConn() driver.Conn {
 	return client.conn
 }
+
+func GetSessionsQueryImpl(admin *model.Admin, params modelInputs.QueryInput, projectId int, retentionDate time.Time, selectColumns string, groupBy *string, orderBy *string, limit *int, offset *int) (string, []interface{}, bool, error) {
+	sb := sqlbuilder.NewSelectBuilder()
+	sb.From(SessionsJoinedTableConfig.TableName)
+	sb.Select(selectColumns)
+
+	sb.Where(sb.LessEqualThan("CreatedAt", params.DateRange.EndDate)).
+		Where(sb.GreaterEqualThan("CreatedAt", params.DateRange.StartDate))
+
+	parser.AssignSearchFilters(sb, params.Query, SessionsJoinedTableConfig)
+	if groupBy != nil {
+		sb = sb.GroupBy(*groupBy)
+	}
+	if orderBy != nil {
+		sb = sb.OrderBy(*orderBy)
+	}
+	if limit != nil {
+		sb = sb.Limit(*limit)
+	}
+	if offset != nil {
+		sb = sb.Offset(*offset)
+	}
+
+	sql, args := sb.BuildWithFlavor(sqlbuilder.ClickHouse)
+
+	// TODO(spenny): custom rules to check - sampleField
```

Comment:
TODO(spenny): figure out with Julian the best way to add a new sample with the new search UI, and apply here